core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,233 @@
+import org.bukkit.event.Listener;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchFinishEvent;
+import tc.oc.pgm.api.match.factory.MatchModuleFactory;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.api.player.PlayerRelation;
+import tc.oc.pgm.api.player.event.MatchPlayerDeathEvent;
+import tc.oc.pgm.events.ListenerScope;
+import tc.oc.pgm.tracker.damage.ProjectileInfo;
+import tc.oc.util.bukkit.component.Component;
+import tc.oc.util.bukkit.component.ComponentUtils;
+import tc.oc.util.bukkit.component.Components;
+import tc.oc.util.bukkit.component.types.PersonalizedText;
+import tc.oc.util.bukkit.component.types.PersonalizedTranslatable;
+import tc.oc.util.bukkit.nms.NMSHacks;
+
+@ListenerScope(MatchScope.RUNNING)
+public class StatsMatchModule implements MatchModule, Listener {
+
+  public static class Factory implements MatchModuleFactory<StatsMatchModule> {
+
+    @Override
+    public StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+      return new StatsMatchModule();
+    }
+  }
+
+  public static class playerStats {
+    private int kills = 0;
+    private int deaths = 0;
+    private int killstreak = 0;
+    private int killstreakMax = 0;
+    private int longestBowKill = 0;
+
+    public void onMurder() {
+      kills++;
+      killstreak++;
+      if (killstreak > killstreakMax) killstreakMax = killstreak;
+    }
+
+    public void onDeath() {
+      deaths++;
+      killstreak = 0;
+    }
+
+    public void setLongestBowKill(double distance) {
+      if (distance > longestBowKill) {
+        longestBowKill = Integer.parseInt(new DecimalFormat("#").format(Math.round(distance)));
+      }
+    }
+
+    public Component getBasicStatsMessage() {
+      String KD;
+      if (deaths == 0) {
+        KD = "0";
+      } else {
+        DecimalFormat df = new DecimalFormat("#.##");
+        KD = df.format(kills / deaths);
+      }
+      return new Component(
+          new PersonalizedTranslatable(
+                  "stats.basic",
+                  new PersonalizedText(Integer.toString(kills), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(killstreak), ChatColor.GREEN),
+                  new PersonalizedText(Integer.toString(deaths), ChatColor.RED),
+                  new PersonalizedText(KD, ChatColor.GREEN))
+              .render());
+    }
+  }
+
+  public static Map<String, playerStats> allPlayerStats;
+
+  public StatsMatchModule() {
+    allPlayerStats = new HashMap<>();
+  }
+
+  @EventHandler
+  public void onPlayerDeath(MatchPlayerDeathEvent event) {
+    MatchPlayer victim = event.getVictim();
+    MatchPlayer murderer = null;
+    try {
+      murderer = event.getKiller().getParty().getPlayer(event.getKiller().getId());
+    } catch (NullPointerException ignored) {
+    }
```

Comment:
You shouldn't be catching `NPE`s :D If something returns `null` please check it with the `if` statements